src/nvidia_resiliency_ext/shared_utils/logger.py

```
@@ -0,0 +1,584 @@
+        self._send_message_to_aggregator(log_msg)
+
+    def _send_message_to_aggregator(self, log_msg: LogMessage):
+        """Send a message to the aggregator using file-based communication."""
+        # Create message directory in temp dir
+        msg_dir = os.path.join(self._temp_dir, "pending_messages")
+        os.makedirs(msg_dir, exist_ok=True)
+
+        # Use a single message file per rank
+        msg_file = os.path.join(msg_dir, f"rank_{self._local_rank}.msg")
+
+        try:
+            # Append message to the rank's message file
+            with open(msg_file, 'a') as f:
+                f.write(
+                    f"{log_msg.level}\t{log_msg.message}\t{log_msg.rank}\t{log_msg.local_rank}\t{log_msg.hostname}\t{log_msg.timestamp}\n"
+                )
+                f.flush()  # Ensure message is written immediately
+
+        except Exception as e:
+            # If file-based communication fails, just log to stderr
+            sys.stderr.write(f"Failed to send message to aggregator: {e}\n")
+            sys.stderr.flush()
+
+    def _check_pending_messages(self):
+        """Check for pending messages from other ranks (aggregator only)."""
+        if not self._is_aggregator:
+            return
+
+        msg_dir = os.path.join(self._temp_dir, "pending_messages")
+        if not os.path.exists(msg_dir):
+            return
+
+        try:
+            # Look for message files from all ranks (including this aggregator rank)
+            for filename in os.listdir(msg_dir):
+                if filename.startswith('rank_') and filename.endswith('.msg'):
+                    msg_file = os.path.join(msg_dir, filename)
+                    try:
+                        # Get current file size
+                        try:
+                            file_size = os.path.getsize(msg_file)
+                        except (IOError, OSError):
+                            # File doesn't exist or can't be accessed, skip
```

If it doesn't exist, that means either the mounts are not visible across processes or that something went wrong.

This should at minimum log a warning to the job-level stderr.

---

Change all the exception handling in the logger to output an error rather than silently suppressing.

---

Logs are emitted in exception path